### PR TITLE
Sync pay-server changes to sorbet-runtime

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Prep work for using sorbet-runtime in pay-server.

### Test plan

Existing tests.

### CODE REVIEW

File: gems/sorbet-runtime/lib/types/private/abstract/declare.rb [hunk 1]
```diff
@@ -18,10 +18,6 @@ def self.declare_abstract(mod, type:)
     mod.extend(T::InterfaceWrapper::Helpers)
 
     if mod.is_a?(Class)
-      if mod < T::Struct
-        raise "#{mod.name} is a subclass of T::Struct and cannot be declared abstract"
-      end
-
```
Comment (jez) [inline gems/sorbet-runtime/lib/types/private/abstract/declare.rb]: @azdavis can you confirm that you do not want these checks yet?

They were deleted in pay-server, but not deleted in Sorbet.

File: gems/sorbet-runtime/test/types/struct.rb [hunk 2]
```diff
@@ -9,14 +9,4 @@ class Opus::Types::Test::StructValidationTest < Critic::Unit::UnitTest
...
-
-  it "forbids declaring a Struct as abstract" do
-    err = assert_raises(RuntimeError) do
-      Class.new(T::Struct) do
-        extend T::Helpers
-        abstract!
-      end
-    end
-    assert_includes(err.message, "is a subclass of T::Struct and cannot be declared abstract")
-  end
```
Comment (jez) [inline gems/sorbet-runtime/test/types/struct.rb]: @azdavis same for this one

File: gems/sorbet-runtime/lib/types/props/_props.rb [hunk 3]
```diff
@@ -87,7 +87,7 @@ def reload_decorator!; @decorator = decorator_class.new(self); end
     #   hash. This will not affect objects loaded by {.from_hash}.
     # @option rules [Proc] :factory A `Proc` that will be called to
     #   generate an initial value for this prop on {#initialize}, if
-    #   none is provided.
+    #   none is providede.
```
Comment (ptarjan) [inline gems/sorbet-runtime/lib/types/props/_props.rb]: this is a regression?

Comment (jez) [reply gems/sorbet-runtime/lib/types/props/_props.rb]: whoops yes

File: gems/sorbet-runtime/lib/types/private/abstract/declare.rb (hunk 1 above)
Comment (azdavis) [reply gems/sorbet-runtime/lib/types/private/abstract/declare.rb]: Last I checked, pay-server breaks if this is not removed.

File: gems/sorbet-runtime/test/types/struct.rb (hunk 2 above)
Comment (azdavis) [reply gems/sorbet-runtime/test/types/struct.rb]: If we take out the check, then we should take out the test testing to see if the check is there, so yes.

**Comment by DarkDimius** [comment]
rebased

**Pull request merged**